### ISSUE TITLE
Compiler warning when compiling with clang++

### ISSUE THREAD

**Issue by kkarbowiak** [VROOM-Project/vroom#699]
When I compile vroom using clang++ (command `make CXX=clang++`), the compiler outputs a warning:
```
structures/vroom/input/input.cpp:681:41: warning: braces around scalar initializer [-Wbraced-scalar-init]
            d_m->second = Matrix<Cost>({{0}});
                                        ^~~
```
I believe this is a valid warning as the `Matrix`'s constructor accepts a scalar and the braces are redundant.

For the record, I'm using clang version 13.0.1.

**Comment by jcoupey** [comment]
I think this is a left-over from the time where matrices where vectors of vectors and the initializer list made sense. This was changed back in #491 and it looks like we introduced a regression back then.

Now `Matrix<Cost>({{0}})` behaves like `Matrix<Cost>(0)` which creates an empty matrix, not a one-element matrix with a zero. I confirmed the problem by creating a single-location problem and got an explicit `[Error] location_index exceeding matrix size for car profile.`.
